unable to push

There's a workflow A:
```yml
name: Root workflow

on: [push]

jobs:
  call-build:
    permissions:
      contents: write
    uses: org/workflows/.github/workflows/reusable-B.yml@master
    secrets: inherit
```

that calls a reusable workflow `reusable-B.yml`:
```yml
jobs:
  version-tag-changelog:
    name: Version & tag the release
    if: ${{ github.ref == 'refs/heads/master' }}
    runs-on: ubuntu-latest
    steps:
    - uses: actions/create-github-app-token@v2
      id: app-token
      with:
        app-id: ${{ vars.APP_ID }}
        private-key: ${{ secrets.APP_KEY }}
    - uses: actions/checkout@v5
      with:
        token: ${{ steps.app-token.outputs.token }}
        ref: ${{ github.head_ref }}
        persist-credentials: false
    - name: Configure git
      run: |
        git config --global user.name 'workflow-bot'
        git config --global user.email 'ci@github'
    - name: Install zestreleaser & cut a tag
      env:
        GITHUB_TOKEN: ${{ steps.app-token.outputs.token }}
      run: |
        some commands that create commits and tags
        git push --follow-tags
```

Yet the last push fails with 128:
```
fatal: could not read Username for 'https://github.com/': No such device or address
Error: Process completed with exit code 128.
```

[This discussion](https://github.com/orgs/community/discussions/26580#discussioncomment-12894890) suggests `url` needs to be reconfigured to contain the token:
```sh
git config --global url."https://x-access-token:${{ steps.app-token.outputs.token }}@github.com/".insteadOf https://github.com/
```

Should this instruction be included in the [readme](https://github.com/actions/create-github-app-token?tab=readme-ov-file#configure-git-cli-for-an-apps-bot-user)?

--- a/README.md
+++ b/README.md
@@ -104,6 +104,7 @@ jobs:
           # required
           client-id: ${{ vars.APP_CLIENT_ID }}
           private-key: ${{ secrets.APP_PRIVATE_KEY }}
+          permission-contents: write
       - name: Get GitHub App User ID
         id: get-user-id
         run: echo "user-id=$(gh api "/users/${{ steps.app-token.outputs.app-slug }}[bot]" --jq .id)" >> "$GITHUB_OUTPUT"
@@ -112,7 +113,10 @@ jobs:
       - run: |
           git config --global user.name '${{ steps.app-token.outputs.app-slug }}[bot]'
           git config --global user.email '${{ steps.get-user-id.outputs.user-id }}+${{ steps.app-token.outputs.app-slug }}[bot]@users.noreply.github.com'
-      # git commands like commit work using the bot user
+          gh auth setup-git
+        env:
+          GH_TOKEN: ${{ steps.app-token.outputs.token }}
+      # git commands like commit and push work using the bot user
       - run: |
           git add .
           git commit -m "Auto-generated changes"

